ansible/library/extract_log.py

```diff
@@ -90,9 +90,16 @@ def extract_line(directory, filename, target_string):
         file = gzip.GzipFile(path)
     else:
         file = open(path)
-    result = None
+    result = []
     with file:
-        result = [(filename, line) for line in file if target_string in line and 'nsible' not in line]
+        for line in file:
+            # This might be a gunzip file issue, there has been '\x00's in front of the
+            # log entry timestamp which messes up with the comparator.
+            # Prehandle lines to remove these sub-strings
```

Comment:
Are you sure these zeros from gzip file?
I suspect rotate logs procedure rotate syslog file while we're reading it.
I'd better to stop log rotator while we reading from syslog files.

Reply:
Maybe it came from logrotate, but the thing is that I found these entries in old log files, not recent ones. That means if problem happened, it will stay there. Stopping log rotating while reading log won't fix the old log files.

Reply:
Another alternative is to rm all *.gz before test. That doesn't guarantee newly rotated gz files never hit these issue during test though.

Reply:
We can't delete .gz files. Our default syslog file size is small. If the reboot was normal start of the reboot is in syslog.1
If we got any issues, you can find start of the reboot in syslog.2.gz or earlier.

Reply:
I think we have to disable log rotate while we're running the script. Otherwise we could miss some information.
Also, I don't understand which component save zeros to syslog file and why. Generally I don't like to fix something, while I don't figure out source of the issue. Like in this case.
Also you fix could be only this long
result = [(filename, line) for line in file if target_string in line.replace('\x00', '') and 'nsible' not in line]

Reply:
That is a much better format. Thanks Pavel!